staging/src/k8s.io/kubectl/pkg/cmd/explain/explain.go

```diff
@@ -51,6 +51,9 @@ var (
 		# Get all the fields in the resource
 		kubectl explain pods --recursive
 
+		# Get all the fields in the resource with their descriptions
+		kubectl explain pods --versobe-recursive
```

Comment:
Mentioned the depth in this comment https://github.com/kubernetes/kubernetes/pull/124218#discussion_r1556216753

Reply:
So this is being added because we already have 2 implicit modes:

1. Long-form
2. Recursive

However we infer the value of the long form view based on the value of recursive. This PR seems to aim to separate the concerns. I think for this case verbose may be overloaded to cause confusion with extended logging output.  Maybe another flag is justified, so we could have 4 modes:

1. Recursive Long
2. Recursive Short
3. Shallow Long
4. Shallow Short

To keep behavior the same however, we'd need to adopt the uncommon/possibly confusing pattern of defaulting the value of a new long-form flag based on the value of the recursive flag. I'm not sure how to best mitigate that

Reply:
Should this be a seperate template to be honest?

Reply:
> Should this be a seperate template to be honest?

I think not, the template already has a concept of long vs short display along with recursion but the separation doesn't bubble to the interface. Adding this doesnt introduce much complexity IMO

Reply:
>  I don't think it has a lot of value does it?

Unfortunately we dont have a lot of data on who is using it and in the absense of that Id be uncomfortable taking this away. Some might find it convenient and easy to read to list out all the fields for a quick reminder 🤷🏽